### DISCUSSION TITLE
How to make hosted Outline app public/open for everyone?

### DISCUSSION THREAD

**Discussion by shtefcs** [outline/outline#2412] [General]
First of all, wonderful work. Simple, clean, and powerful. Exactly what I was looking for as an open-source project for a while.

We have self-hosted Outline, and I am using it to create documentation for our project Automatio.co. 

My question is how to make everything publicly available, and not locked under the registration gate? I mean on Collections and all documents under it?

We want to use Outline as public documentation, so everyone can access it and be indexable by search engines.

Is this possible with Outline? I can see its a perfect platform for such a thing, except I can't figure it out how to make everything public.

**Comment by balupton** [comment]
Good thing I found this, as it somewhat matches my use case for mixing and matching public and private content, for a personal knowledge base of which some pages are shared with peers and some are completely public.

Keen to hear the answer. If there isn't an official solution, considering their expansive API, perhaps one could setup a cloudflare worker or static generator that renders the "public" pages publicly.

**Comment by tommoor** [comment]
> Is this possible with Outline? I can see it's a perfect platform for such a thing, except I can't figure it out how to make everything public.

It's not currently possible but we plan on building out this functionality in the near future. In the meantime someone could definitely develop a solution based on the API.

**Reply by shtefcs** [reply]
> > Is this possible with Outline? I can see it's a perfect platform for such a thing, except I can't figure it out how to make everything public.
> 
> It's not currently possible but we plan on building out this functionality in the near future. In the meantime someone could definitely develop a solution based on the API.

So many people will need the same, and I am not really sure why you are missing such a big chunk of users, just by having that simple feature? Its been a while since people searched, it's not a new concept.

**Reply by benjamin-carriou** [reply]
+1

**Comment by onny** [comment]
Maybe we could open an issue to keep track on the progress of this implementation. I'm also very interested in this!

**Comment by aresstokrat** [comment]
That's definitely a gem feature to have.

**Comment by piyushdatazip** [comment]
Waiting for this feature, Is there a public roadmap for Outline @tommoor ?

**Reply by d41ly** [reply]
In the current version of Outline you can publicly share your collections and make them available to everyone, without registration.

**Reply by piyushdatazip** [reply]
Actually we're using Outline for Customer documentation, currently we're doing the same as you said, sharing links. and waiting for this as this will help in exposing all docs at home url.

**Reply by keithfawcett** [reply]
Making individual docs in Collections public is not ideal. There is no sidebar for these so it's nearly impossible to make a public site with Outline.

**Reply by tommoor** [reply]
There is a sidebar, this is how our own help guides are published. 

Collection sharing is next on the roadmap though,currently rebuilding collection permissions UI to allow for this

**Reply by keithfawcett** [reply]
ok, I figured it out, but it wasn't easy to understand. I had to create a Collection, then create a doc, then create a nested doc. Only then will you get a sidebar. 

If you only have a Collection and doc, without a nested doc, then the sidebar doesn't show (but the space is there, it's just missing the logo, search field, and document structure).

**Reply by mrchrisadams** [reply]
@keithfawcett thank you for this last comment. It really helped me just now with trying to figure out how to share some pages from our instance of Outline.

@tommoor I have one question:

> There is a sidebar, this is how our own help guides are published.

When you refer to the guides, you're referring to this guide here? 

https://docs.getoutline.com/s/guide

Is the guide literally just a set of nested documents, or is there anything custom going on there?

**Reply by Limezy** [reply]
As far as I understood there is nothing custom, it's a unique document "Guide" with two levels of nested documents. You coud do the exact same with your own instance.

**Reply by mohamed-benoba** [reply]
Is there any updates to this? @tommoor

**Reply by tommoor** [reply]
We have begun development of #1160 as of this week which goes a long way towards it

**Reply by Limezy** [reply]
That is a great piece of news !

**Comment by Novapixel1010** [comment]
This is now available as of [v0.86.0](https://github.com/outline/outline/releases/tag/v0.86.0) for even more details check #1160

**Comment by sokai** [comment]
Hey!

In addition to #1160 – which I really appreciate! – I would add the following/similar discussions:

* #10190
* #5124

KR
